x-pack/plugins/security_solution/server/lib/timeline/saved_object/pinned_events/index.test.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FrameworkRequest } from '../../../framework';
+
+import { getAllPinnedEventsByTimelineId, PINNED_EVENTS_PER_PAGE } from '.';
+
+describe('pinned events', () => {
+  let mockFindSavedObject: jest.Mock;
+  let mockRequest: FrameworkRequest;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    mockFindSavedObject = jest.fn().mockResolvedValue({ saved_objects: [], total: 0 });
+    mockRequest = {
+      user: {
+        username: 'username',
+      },
+      context: {
+        core: {
+          savedObjects: {
+            client: {
+              find: mockFindSavedObject,
+            },
+          },
+        },
+      },
+    } as unknown as FrameworkRequest;
```

Comment:
> I haven't come across this `as x as y` notation yet. Is there a term for this? Would love to look it up in the TS documentation.

When [TypeScript determines if a single assertion is not enough](https://basarat.gitbook.io/typescript/type-system/type-assertion#how-typescript-determines-if-a-single-assertion-is-not-enough), casting to `as unknown` first is essentially a form of type erasure, because `unknown` is a [top type](https://2ality.com/2020/06/any-unknown-typescript.html) that guarantees the assertion that follows it will succeed, because whatever follows is guaranteed to be a subtype of `unknown`.

The technique used in the new unit test to mock a `FrameworkRequest` (that required the double assertion) is actually borrowed from [these existing tests](https://github.com/elastic/kibana/blob/1c0ba2ffae8e749c417b5d42da0d38f154bd3a29/x-pack/plugins/security_solution/server/lib/timeline/saved_object/timelines/index.test.ts#L80).

Reply:
Thanks! TIL!